### PR TITLE
Add fetching from background in content util

### PR DESCRIPTION
**Pull request by yayuyokitano**
Allows connectors to easily run fetch requests through the service worker.
The main use of this is allowing connectors to ignore CORS restrictions when fetching from outside sources.
Implemented as a contributor has expressed interested in an implementation of NTS that uses fetch requests to a domain with CORS restrictions.

### CODE REVIEW

**Review by inverse — APPROVED** [review]
Seems reasonable :+1:

File: src/core/content/util.ts [hunk 1]
```diff
@@ -1072,3 +1073,77 @@ export interface ChannelInfo {
...
+	switch (type) {
+		case 'text':
+			return {
+				ok: true,
+				content: res.content,
+			};
+		case 'json':
+			return {
+				ok: true,
+				content: JSON.parse(res.content) as unknown,
```
[the remark this answers is not in the document]
Comment (yayuyokitano) [reply src/core/content/util.ts:1138]: [typescript unknown type](https://mariusschulz.com/blog/the-unknown-type-in-typescript)

TL;DR: anything can safely be assigned _to_ an unknown type variable, but to actually use it you have to either narrow it or explicitly cast it first. It’s essentially like a safer version of any.
